File: src/main/kotlin/microserver/bio/listener/ServerListener.java
package microserver.bio.listener;


import microserver.bio.MicroConnection;
import microserver.bio.ServerConfig;
import microserver.bio.ServerState;

public abstract class ServerListener {

    public abstract void onCreate(ServerConfig config, ServerState state);

    public abstract void onStart(ServerConfig config, ServerState state);

    public abstract void onAccept(final MicroConnection connection);

    public abstract void onClose(ServerConfig config, ServerState state);

}

File: src/main/kotlin/microserver/bio/listener/ServerListener.kt
package microserver.bio.listener;


import microserver.bio.MicroConnection
import microserver.bio.ServerConfig
import microserver.bio.ServerState

abstract class ServerListener {

    abstract fun onCreate(config: ServerConfig, state: ServerState);

    abstract fun onStart(config: ServerConfig, state: ServerState);

    abstract fun onAccept(connect: MicroConnection)

    abstract fun onClose(config: ServerConfig, state: ServerState)

}

